Automattic/wp-calypso
# Display an UpgradeModal on dotorg themes

Related to #81089, branched from #81516 (which is the merge target purely to make review easier)

## Proposed Changes

 On a community (dotorg) theme sheet display a button to activate the theme rather than just a banner prompting purchase of the business plan.
    
This button will display a modal (if necessary) extolling the virtues of a business plan and explaining that it is a prerequisite to using the theme. This is the same behaviour as partner, woo themes etc in #81516 but with slightly different text.
    
To facilitate this the ThemeUpgradeModal uses the `getCanonicalTheme` selector, which means the `useThemeDetails` hook is obsolete and no longer used.

### Remaining tasks

- [ ] Make it not display a message about "Your plan doesn't support this" when clicking on the primary modal button
- [ ] Return the user back to the theme page where they will be able to activate the theme (smoothing this is #81096)

## Testing Instructions



 1. Use calypso live branch
 2. Visit /themes on a < business plan site
 3. Search for something to make community (dotorg) themes show up e.g. 'Astra'
 4. Click one of the community themes
 5. See a 'Activate this design' button appears unlike on live
 6. Press the button
 7. See a modal appear telling you stuff about the business plan
 8. Press 'Upgrade plan'
 9. Go to checkout to buy the plan
 10. Finish checkout, find yourself on the theme page ready to install the theme

## Pre-merge Checklist
